storybook/src/pages/amsterdam/common/AppHeader.tsx

```diff
@@ -1,13 +1,74 @@
-import { Grid, Header, PageMenu } from '@amsterdam/design-system-react'
-import { Alignment as PageMenuStory } from '../../../components/PageMenu/PageMenu.stories'
-
-export const AppHeader = () => (
-  <Grid>
-    <Grid.Cell span="all">
-      <Header
-        links={<PageMenu alignEnd>{PageMenuStory.args?.children}</PageMenu>}
-        menu={<button className="ams-header__menu-button">Menu</button>}
-      />
-    </Grid.Cell>
-  </Grid>
-)
+/**
+ * @license EUPL-1.2+
+ * Copyright Gemeente Amsterdam
+ */
+
+import { Grid, Header, PageMenu, SearchField } from '@amsterdam/design-system-react'
+import { MenuIcon, SearchIcon } from '@amsterdam/design-system-react-icons'
+import { useMemo, useState } from 'react'
+import { Default as MegaMenuStory } from '../../../components/MegaMenu/MegaMenu.stories'
+
+type Feature = 'menu' | 'search'
+
+export const AppHeader = () => {
+  const [visibleFeature, setVisibleFeature] = useState<Feature | undefined>(undefined)
+
+  const toggleFeature = (feature: Feature) =>
+    setVisibleFeature((currentFeature) => (feature === currentFeature ? undefined : feature))
+
+  const visibilityClass = (feature: Feature) => (feature === visibleFeature ? null : 'ams-visually-hidden')
```

Comment:
Is `ams-visually-hidden` the way to go? Yes you'll hide it but for the visually impaired, they will get the menu read out loud to them everytime. It should probably just use `display: none` and the menu (toggle) button should have `aria-controls="mega-menu-id"` so that the menu gets read to them when they toggle the menu.

Reply:
Even Copilot agrees:

Using "visually hidden" to hide the megamenu may be a poor decision due to the following reasons:

1. Accessibility Issues: "Visually hidden" is intended for content that is hidden from sighted users but accessible to screen readers. Using it to hide interactive elements like a megamenu can confuse screen reader users as they might access the menu without visual context.
2. User Experience: Sighted users may face unexpected behavior if the megamenu is hidden visually but still interactable. This can lead to confusion and a poor user experience.
3. Best Practices: The repository guidelines suggest visually available content should also be accessible to non-visual user agents and vice versa. Hiding the megamenu only visually contradicts this practice.

Reply:
Pretty cool it even refers to your own documentation

Reply:
Correct, I haven’t implemented all accessibility features yet; I wanted to get the team’s feedback on this approach and API first. We will ensure that both sighted and non-sighted people can navigate properly. Screen reader users will want to hear the items, or they can use the Skip Link.

Someone suggested the Mega Menu must be in the DOM for search engines. That could well be – we should research that. Of course, we could employ other strategies like XML sitemaps, but I don’t expect they’ll allow us to hide the main menu for crawlers entirely.

I do remove the Search bar from the DOM, so I have considered it.

> Pretty cool it even refers to your own documentation

Yes, it is!

Reply:
Search engines execute javascript nowadays and are able to crawl the site without the help of the navigation. The child links of the Google search results are based on relevance, not the sitemap or main nav. Forcing visually impaired users to always use the skip link is not the way to go.